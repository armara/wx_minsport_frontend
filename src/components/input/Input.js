/* eslint-disable react/jsx-props-no-spreading */
import React, { forwardRef } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import 'components/input/Input.scss'

const Input = forwardRef(
  (
    { type, placeholder, value, onChange, className, input, disabled, ...rest },
    ref
  ) => (
    <input
      type={type}
      placeholder={placeholder}
      value={value}
      onChange={onChange}
      ref={ref}
      disabled={disabled}
      className={classnames([className, 'base-input', 'input'])}
      {...input}
      {...rest}
    />
  )
)

Input.defaultProps = {
  type: 'text',
  placeholder: '',
  onChange: () => {},
  className: '',
  input: {},
  disabled: false,
}

Input.propTypes = {
  disabled: PropTypes.bool,
  type: PropTypes.oneOf(['text', 'checkbox', 'time']),
  input: PropTypes.shape({}),
  placeholder: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  onChange: PropTypes.func,
  className: PropTypes.string,
}

export default Input
