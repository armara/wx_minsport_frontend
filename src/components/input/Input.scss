@import "styles/mixins";

.form-input {
  @include form-input();

  &::placeholder {
    color: $red-gray;
  }
}