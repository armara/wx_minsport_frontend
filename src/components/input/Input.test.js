import React from 'react'
import Input from 'components/input/Input'
import { configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

configure({ adapter: new Adapter() })

describe('<Input />', () => {
  const testInputValue = 'hohenzollern'
  const inputWrapper = mount(<Input value={testInputValue} />)
  const findInput = (wrapper = inputWrapper) => wrapper.find('.base-input')

  it('renders correctly default props', () => {
    const tree = renderer.create(<Input value={testInputValue} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('is stateless component', () => {
    const renderedInput = mount(<Input value={testInputValue} />)
    const instance = renderedInput.instance()

    expect(instance).toEqual(null)
  })

  it('is selectable with input tag', () => {
    expect(findInput().length).toBe(1)
  })
})
