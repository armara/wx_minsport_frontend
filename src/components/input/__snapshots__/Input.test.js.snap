// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly default props 1`] = `
<input
  className="base-input input"
  disabled={false}
  onChange={[Function]}
  placeholder=""
  type="text"
  value="hohenzollern"
/>
`;
